lib/python/rose/macro.py

```diff
@@ -467,14 +467,15 @@ def load_meta_path(config=None, directory=None, is_upgrade=False,
         meta_keys = ["rose-all"]
     else:
         key = opt_node.value
-        if "/" not in key:
-            key = key + "/" + rose.META_DEFAULT_VN_DIR
+        split_key = os.path.split(key)
+        if not split_key[0]:
+            key = os.path.join(key, rose.META_DEFAULT_VN_DIR)
         meta_keys = [key]
         if is_upgrade:
-            meta_keys = [key.split("/")[0]]
+            meta_keys = [os.path.split(key)[0]]
         else:
-            default_key = (key.rsplit("/", 1)[0] + "/" +
-                           rose.META_DEFAULT_VN_DIR)
+            default_key = os.path.join(os.path.split(key)[0],
+                                       rose.META_DEFAULT_VN_DIR)
```

Comment:
The `/` here is part of the syntax, instead of the path separator of the current OS. Your change will just happen to work on a Unix/Linux system, but is not entirely correct.

Have you tested this logic with all the combinations? E.g. I can think of these:
```cfg
meta=foo
meta=foo/HEAD
meta=foo/10.8
meta=foo/bar
meta=foo/bar/HEAD
meta=foo/bar/10.8
```